### DISCUSSION TITLE
Exporting different replays from the same user on the same beatmap overwrites the previous exported one

### DISCUSSION THREAD

**Discussion by integerrr** [ppy/osu#21438] [Q&A] [answered]
As the title suggests, when you export a replay through leaderboard context menu, and then exporting a different replay of the same map from the same user overwrites the previous one, due to the output file names being the same, even though the replay/score is different.

Scores in stable when exported, appends the date which the score was set in the filename, which is what I've tried to mock up locally [here](https://github.com/integerrr/osu/commit/350d67370be9c84d4a572d69de39deadd2f141a9).

One notable difference though, in stable even if the exported filename is the same it appends a `(1)` etc. to the filename whereas in lazer it would be overwritten instead.

**Comment by smoogipoo** [comment]
You should use `NamingUtils.GetNextBestName()` for this.

**Reply by peppy** [reply]
@integerrr are you planning to attempt this one?

**Reply by integerrr** [reply]
Yes but will very likely need help

**Reply by integerrr** [reply]
someone's already PR'd a solution but I'll put my (lots of googling effort) result here for feedback I guess?
https://github.com/integerrr/osu/commits/score-export-with-date

**Comment by peppy** [comment, marked as answer]
Tracking at https://github.com/ppy/osu/issues/21464 so we don't lose track.
